packages/gatsby-mdx/gatsby-node.js

```diff
@@ -13,6 +16,38 @@ exports.onCreateNode = require("./on-create-node");
  */
 exports.setFieldsOnGraphQLNodeType = require("./extend-node-type");
 
+/**
+ * Add frontmatter as page context for MDX pages
+ */
+exports.onCreatePage = async ({ page, actions }, pluginOptions) => {
+  const { createPage, deletePage } = actions;
+  const { extensions, ...options } = defaultOptions(pluginOptions);
+  const ext = path.extname(page.component);
+
+  if (extensions.includes(ext)) {
+    const content = await fs.readFile(page.component, "utf8");
+    const code = await mdx(content, options);
+
+    // grab the exported frontmatter
+    const { frontmatter } = extractExports(code);
+
+    deletePage(page);
+    createPage(
+      merge(
+        {
+          context: {
+            frontmatter: {
+              title: ``, // always include a title
```

Comment:
why always include a title? Seems like we're assuming a blog post content type or something

Reply:
I was going for consistency with the Mdx node. I don't have a particular use case in mind for it, so I'll take it out.